fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/AvailabilityTestingUtils.groovy

```diff
@@ -62,4 +64,21 @@ class AvailabilityTestingUtils extends Specification {
                     table.setAvailability(new ConcreteAvailability(table.getTableName(), table.getSchema().getColumns(), new TestDataSourceMetadataService(allIntervals)));
                 }
     }
+
+    /**
+     * Construct a map of Column object to its all available intervals
+     *
+     * @param map a map of Column in String representation to its all available intervals in String representations
+     *
+     * @return a map of Column object to its all available intervals
+     */
+    static def constructColumnToIntervalMap(Map<String, Set<String>> map) {
+        return map.entrySet().stream()
+                .collect(
+                    Collectors.toMap(
+                            {Map.Entry it -> new Column(it.key)},
```

Comment:
It's not a cast in this context, it's a type indication, but otherwise you are correct, Groovy's dynamic typing means that it's not needed here (unless we wanted it for clarity)